libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -250,9 +251,12 @@ bool ASTJsonConverter::visit(ImportDirective const& _node)
 	std::vector<pair<string, Json::Value>> attributes = {
 		make_pair("file", _node.path()),
 		make_pair("absolutePath", _node.annotation().absolutePath),
-		make_pair(m_legacy ? "SourceUnit" : "sourceUnit", nodeId(*_node.annotation().sourceUnit)),
 		make_pair("scope", idOrNull(_node.scope()))
 	};
+
+	if (_node.annotation().sourceUnit)
```

Comment:
When does it not have a source unit?

Reply:
Pretty much always when we do stopAfter=parsing

Reply:
Could also use `idOrNull`